internal/logger/logger.go

```diff
@@ -46,94 +41,78 @@ var NullLogger = nullLogger{}
 
 var (
 	logger Logger = NullLogger
-	lock   sync.Mutex
```

Comment:
@benhoyt I still don't believe we can remove this, its important to protect the `logger` variable's memory.

Reply:
I'm not sure I follow what the concern is: multiple goroutines *reading* from a variable concurrently is fine, it's only when there's something writing as well that it's unsafe. `SetLogger` is the only thing that writes it, but that's only called once at startup from the main goroutine, which I've noted in `SetLogger`'s doc comment (`MockLogger` also calls `SetLogger`, but we not running tests concurrently within the package, so that's fine).

I removed this because it seems a shame to be locking something that never needs it, as we're never calling `SetLogger` except the once on startup. And now that the Logger type locks itself (the stdlib's logger did as well), it would have been two lock/unlock sequences for every write.